cadence/contracts/flix_registry.cdc

```diff
@@ -0,0 +1,172 @@
+access(all) 
+contract FLIXRegistry {
+
+    access(all)
+    event ContractInitialized()
+
+    access(all)
+    event Published(alias: String, id: String, cadenceBodyHash: String)
+
+    access(all)
+    event Removed(id: String)
+
+    access(all)
+    event Deprecated(id: String)
+
+    access(all)
+    event AliasLinked(alias: String, id: String)
+
+    access(all)
+    event AliasUnlinked(alias: String)
+
+    access(all)
+    enum FLIXStatus: UInt8 {
+
+        access(all)
+        case active
+
+        access(all)
+        case deprecated
+    }
+
+    access(all)
+    struct FLIX {
+        access(all)
+        let id: String
+
+        access(all)
+        let jsonBody: String
+
+        access(all)
+        let cadenceBodyHash: String
+
+        pub(set)
+        var status: FLIXStatus
+
+        init(id: String, jsonBody: String, cadenceBodyHash: String) {
+            self.id = id
+            self.jsonBody = jsonBody
```

Comment:
Would be good if we can include all parts of the data structure inside the FLIX struct, in order to do things like ID regeneration etc.